Paragraph/Primary/index.jsx

```diff
@@ -54,11 +54,16 @@ Primary.propTypes = {
 }
 
 export default compose(
-  themeable((customizations, props) => ({
-    style: {
-      ...props.style,
-      color: customizations.color_text
+  themeable((customizations, props) => {
+    if (['error', 'warning'].includes(props.color)) {
```

Comment:
Some reuse should definitely be done, I will look into it. However I think that we cannot avoid having the check for whether to theme or not inside the `adapter` function, given that it is there were we have access to the `props`...